src/GafferUI/GraphGadget.cpp

```diff
@@ -1065,11 +1177,6 @@ bool GraphGadget::dragEnter( GadgetPtr gadget, const DragDropEvent &event )
 	if( m_dragMode == Moving )
 	{
 		calculateDragSnapOffsets( m_scriptNode->selection() );
-
-		V2f pos = V2f( i.x, i.y );
-		offsetNodes( m_scriptNode->selection(), pos - m_lastDragPosition );
-		m_lastDragPosition = pos;
- 		requestRender();
```

Comment:
I guess if we're not moving the nodes, then we don't need to update it so the delta is correct in `dragMove`, but then do we need to do it for `Select`?

Reply:
> I guess if we're not moving the nodes, then we don't need to update it so the delta is correct in dragMove

Precisely.

> but then do we need to do it for Select

I guess I could have simplified this as well, but I was focussing on the drag move alone. With the code as it stands, we do need to update `m_lastDragPosition` in `Select` mode, because we call `requestRender()` too, and the render we will do is dependent on `m_lastDragPosition`.

> how come we do do the snap offset update?

That's an initialisation rather than an update. We only need to calculate the snap offsets once per drag, so `dragEnter` is the place to do it.

Reply:
Simplified the selection code path a bit in 740f6df.

Reply:
Thanks for cleaning up Selection, just helps when you come at it fresh...

> > how come we do do the snap offset update?
> 
> That's an initialisation rather than an update. We only need to calculate the snap offsets once per drag, so `dragEnter` is the place to do it.

I guess it just seemed odd that the other initialisation is done in `dragBegin`, ie `m_dragStartPosition` etc... no biggie though.